util/syspolicy/policytest/policytest.go

```diff
@@ -83,10 +85,22 @@ func (c Config) GetDuration(key pkey.Key, defaultVal time.Duration) (time.Durati
 
 func (c Config) GetPreferenceOption(key pkey.Key, defaultVal ptype.PreferenceOption) (ptype.PreferenceOption, error) {
 	if val, ok := c[key]; ok {
-		if p, ok := val.(ptype.PreferenceOption); ok {
-			return p, nil
+		switch val := val.(type) {
+		case ptype.PreferenceOption:
+			return val, nil
+		case string:
+			switch val {
+			case "never":
+				return ptype.NeverByPolicy, nil
+			case "always":
+				return ptype.AlwaysByPolicy, nil
+			case "user-decides":
+				return ptype.ShowChoiceByPolicy, nil
+			}
+			panic(fmt.Sprintf("key %s has unrecognized string value %q for ptype.PreferenceOption", key, val))
```

Comment:
I think is is also fine?
```suggestion
			var p PreferenceOption
			err := p.UnmarshalText(([]byte)(val))
			return p, err
```

Reply:
Wow, that accepts all input and never returns an error. Cool. Sure. Done.